src/treelearner/cuda_tree_learner.h

```diff
@@ -0,0 +1,293 @@
+/*!
+ * Copyright (c) 2020 IBM Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ */
+#pragma once
+#ifndef LIGHTGBM_TREELEARNER_CUDA_TREE_LEARNER_H_
+#define LIGHTGBM_TREELEARNET_CUDA_TREE_LEARNER_H_
```

Comment:
This is needed.

Reply:
Could you please elaborate why is it needed?

Reply:
```
cuda_tree_learner.h:43:7: error: redefinition of 'class LightGBM::CUDATreeLearner'
 class CUDATreeLearner: public SerialTreeLearner {
```

Reply:
This is because you have a typo.
```
#ifndef LIGHTGBM_TREELEARNER_CUDA_TREE_LEARNER_H_ 
                        ---^---
#define LIGHTGBM_TREELEARNET_CUDA_TREE_LEARNER_H_
                        ---^---
```

Reply:
Good catch.